### PR TITLE
Various fixes to eliminate errors of multiple types

### PR DESCRIPTION
**Pull request by fidian**
It was possible to have `.trim()` operate on a non-string before, so I
introduced checks to ensure I have a value instead of `undefined.

Fixed the wording in the documentation.

Updated an example script to have the right permissions and to not read
from stdin any longer.

Fixed where request.body may be a Buffer and it should be changed to a
binary string instead of a UTF-8 encoded string (UTF-8 is the default
when typecasting Buffers to strings).

Ignore Elastic Beanstalk CLI config files.

Added deployment instructions.

Updated config to work as the production config.

Added a specialized health check route.

Removed an unnecessary property from the config.

Prevent nginx from logging requests.

Using Mac-acceptable flags for the date command.

### CODE REVIEW

File: doc/deploy.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+3. Decrypt the encryption key.
+
+        gpg --decrypt encryption.key.asc > encryption.key
+
+4. Make sure tests pass.
+
+        npm test
+
+5. If that fails, **stop here**.
```
Comment (AbsentSemicolon) [inline doc/deploy.md]: Should we have a step to say what to do if the tests fail. Something like, "figure out why and fix if possible"?

File: doc/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        --description="Your description for these keys" \
+        --mfa=THE_MFA_SEED --password=REALLY_GOOD_PASSWORD
+
+This will just spit out a set of keys.  They consist of a code and a secret.  The code can be shared with others but keep the secret private!  The secret is used for signing requests.
+
+
+Tokenization, Detokenization
+----------------------------
+
+Signing a request to OpenToken.io is easier than AWS but based on the same principles.  There's a great request to do this for you as well.  Let's tokenize the file `tokenize-me.txt`.
```
Comment (AbsentSemicolon) [inline doc/getting-started.md]: Should add a comma after AWS.

File: doc/deploy.md (hunk 1 above)
Comment (fidian) [inline doc/deploy.md]: I somewhat hope that the person doing the deploy does not need to have that spelled out for them.  Am I assuming too much?

File: lib/signature-ot1.js [hunk 3]
```diff
@@ -59,7 +59,12 @@ module.exports = (accessCodeManager, errorResponse, hash, promise, util) => {
         }
 
         result.push("");
-        result.push(req.body);
+
+        if (req.body) {
+            result.push(req.body.toString("binary"));
```
Comment (AbsentSemicolon) [inline lib/signature-ot1.js]: If this is a buffer, should we use `bufferBinary.toString(req.body)`?

**Comment by AbsentSemicolon** [comment]
Reviewed, raised a few points.

File: doc/getting-started.md (hunk 2 above)
Comment (fidian) [inline doc/getting-started.md]: I've used grammar checkers and they are not saying the same as you.

File: lib/signature-ot1.js (hunk 3 above)
Comment (fidian) [inline lib/signature-ot1.js]: Absolutely!  It was not there when I wrote this code but it certainly is there now.

File: doc/deploy.md (hunk 1 above)
Comment (AbsentSemicolon) [inline doc/deploy.md]: I don't think so, just wanted to know if you thought it should be there, or missed it.

File: doc/getting-started.md (hunk 2 above)
Comment (AbsentSemicolon) [inline doc/getting-started.md]: Very well, guess it just read weird in my head.

**Comment by fidian** [comment]
Addressed the points, fixed the code for one thing.

**Comment by lnagle93** [comment]
Looks good to me.

**Comment by AbsentSemicolon** [comment]
Reviewed and looks good.

**Pull request merged**